src/app/core/entity-details/entity-actions-menu/entity-actions-menu.component.ts

```diff
@@ -66,9 +66,10 @@ export class EntityActionsMenuComponent implements OnChanges {
       this.actions = [];
       return;
     }
-    const allActions: EntityAction[] = this.entityActionsMenuService.getActions(
-      this.entity,
-    );
+    // Pre-filter out bulk-only actions for single-entity context
+    const allActions: EntityAction[] = this.entityActionsMenuService
```

Comment:
for consistency and easier reuse, I am moving this filter to a new method `getActionsForSingle()` in the service